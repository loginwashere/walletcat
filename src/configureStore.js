import { createStore, applyMiddleware, compose } from 'redux'
import reducers from './reducers'
import middleware from './middleware'
import { addLocaleData } from 'react-intl'
import { DEFAULT_LOCALE } from './config'
import localeMessages from './translations'

import enLocaleData from 'react-intl/locale-data/en'
import ruLocaleData from 'react-intl/locale-data/ru'
import ukLocaleData from 'react-intl/locale-data/uk'

addLocaleData(enLocaleData)
addLocaleData(ruLocaleData)
addLocaleData(ukLocaleData)

const locale = DEFAULT_LOCALE
const messages = localeMessages[locale]

const initialState = {
  intl: {
    locale,
    messages,
  }
}

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

const configureStore = (preloadedState = {}) => createStore(
  reducers,
  {
    ...preloadedState,
    ...initialState
  },
  composeEnhancers(applyMiddleware(...middleware))
)

export default configureStore
