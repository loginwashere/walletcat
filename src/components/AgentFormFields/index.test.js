import React from 'react'
import renderer from 'react-test-renderer'
import { reduxForm } from 'redux-form'
import { Provider } from 'react-redux'
import AgentFormFields from '.'
import configureStore from '../../configureStore'

describe('components:AgentFormFields:', () => {
  const AgentFormFieldsDecorated = reduxForm({ form: 'test-form' })(AgentFormFields)

  it('renders without crashing', () => {
    const store = configureStore()
    const tree = renderer.create(
      <Provider store={store} >
        <AgentFormFieldsDecorated />
      </Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

