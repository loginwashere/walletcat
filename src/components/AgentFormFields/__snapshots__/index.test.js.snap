exports[`components:AgentFormFields: renders without crashing 1`] = `
<div>
  <div
    className="form-group has-feedback">
    <label
      className="col-sm-2 control-label"
      htmlFor="form-horizontal-name">
      Name
    </label>
    <div
      className="col-sm-10">
      <input
        autoFocus={undefined}
        className="form-control"
        id="form-horizontal-name"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Name"
        required={true}
        type="text"
        value="" />
    </div>
  </div>
  <div
    className="form-group has-feedback">
    <label
      className="col-sm-2 control-label"
      htmlFor="form-horizontal-description">
      Description
    </label>
    <div
      className="col-sm-10">
      <input
        autoFocus={undefined}
        className="form-control"
        id="form-horizontal-description"
        name="description"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Description"
        required={undefined}
        type="text"
        value="" />
    </div>
  </div>
</div>
`;
